### ISSUE TITLE
table position in Latex

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#2305]
Hi,
I have a bunch of tables that are very nicely aligned in HTML, but got positioned all the way left in Latex. I cannot find a way to tell the Latex writer to put the table in the center, the same we do with images. For instance 

```
.. _doc.shared.dependency_table:

.. table:: Supported versions



   =====================  ============================
   Interface              Supported versions
   =====================  ============================
   |java|                 Sun |java| 1.6+
   |microsoft| |dotnet|   2.1+
   |python| 2             2.6+
   |python| 3             3.1+
   =====================  ============================
```

is rendered

```
\begin{threeparttable}
\capstart\caption{Java, Python and .NET supported versions}\label{changes:doc-shared-dependency-table}\label{changes:id2}
\begin{tabulary}{\linewidth}{|L|L|}
\hline
\textsf{\relax
Interface
} & \textsf{\relax
Supported versions
}\\
\hline
Java
 &  
Sun Java 1.6+
\\
\hline
Microsoft .NET
 &
2.1+
\\
\hline
Python 2
 &
2.6+
\\
\hline
Python 3
 &
3.1+
\\
\hline\end{tabulary}
\end{threeparttable}
```

Am I missing some options? In particular it seems he Latex code is not using a `table` environment to which such placement options could be passed. Would it be a natural extension?

**Comment by jfbu** [comment]
The `threeparttable` environment from the LaTeX package of the same name creates a "box" which can be used "inline" if one wishes (like a very big letter; turns out it has small height and big depth compared to the baseline), or as a "vertical block", but for this, as the documentation of the LaTeX package shows,  the user has to wrap it e.g. in a  `center` or `table` environment. 

Perhaps indeed the LaTeX writer should add this extra wrap-up systematically, as the reStructuredText syntax seems to be for a block-level and not an inline structure.

**Comment by vlotorev** [comment]
I was just going to file the similar issue :)
Actually the issue with tables is more general. A few more examples that are currently cannot be 'styled'  for tables:
- cells background color can't be changed, this is especially useful for changing background color of table header
- header text colour can't be changed
- cells text font can't be changed
- sphinx wrap table header with '\textsf{' (as it can be seen from example above) which also can't be disabled

**Comment by andreacassioli** [comment]
IMHO a minimal setting would need:

1- to wrap a table in a table environment

2-to provide an option, say `:position:`, for the `table` directive in order to specify the table position.

Cell level fine tuning seems to me a step even further and may require a different table package.

**Comment by vlotorev** [comment]
> 2-to provide an option, say :position:, for the table directive in order to specify the table position.

Is it worth upstreaming bug to docutils/ReST syntax to provide `:align:` option in the same way `image` directive has?

**Comment by tk0miya** [comment]
MEMO: related with #1303

**Comment by liminkai520520** [comment]
please use csv-tables

**Comment by tk0miya** [comment]
Now it is implemented at #3405.
Check it out!
